core/src/main/java/org/apache/accumulo/core/file/rfile/bcfile/BoundedRangeFileInputStream.java

```diff
@@ -149,9 +149,13 @@ public boolean markSupported() {
 
   @Override
   public void close() {
-    // Invalidate the state of the stream.
-    in = null;
-    pos = end;
-    mark = -1;
+    final FSDataInputStream inLocal = in;
+    if (inLocal != null) {
+      // synchronize on the FSDataInputStream to ensure we block closing if in the read method
+      synchronized (inLocal) {
+        // Invalidate the state of the stream.
+        in = null;
```

Comment:
The code sets `in` to null to indicate close was called.  This make all the code that's trying to sync on `in` go through a rigmarole.  I think it may simplify the code if a boolean that tracks the closed state was introduced and `in` was never set to null.  Then the code can always safely sync on `in` and check the closed boolean after syncing.